### PR TITLE
[Phi] move reduce_grad kernel into phi

### PR DESCRIPTION
**Pull request by MingMingShangTian**
### PR types

Others
### PR changes

Others
### Describe

Move reduce_mean_grad, reduce_prod, reduce_max_grad, reduce_min_grad kernel into phi.

### CODE REVIEW

**Review by chenwhql — DISMISSED** [review]

File: paddle/phi/kernels/CMakeLists.txt [hunk 1]
```diff
@@ -31,11 +31,12 @@ set(MANUAL_BUILD_KERNELS eigh_kernel gumbel_softmax_kernel gumbel_softmax_grad_k
     matrix_power_kernel matrix_power_grad_kernel maxout_kernel maxout_grad_kernel pool_kernel
     put_along_axis_kernel put_along_axis_grad_kernel segment_pool_kernel segment_pool_grad_kernel
     softmax_kernel softmax_grad_kernel take_along_axis_kernel take_along_axis_grad_kernel
-    triangular_solve_grad_kernel determinant_grad_kernel)
+    triangular_solve_grad_kernel determinant_grad_kernel reduce_kernel)
 kernel_library(eigh_kernel DEPS ${COMMON_KERNEL_DEPS} lapack_function)
 kernel_library(gumbel_softmax_kernel DEPS ${COMMON_KERNEL_DEPS} softmax)
 kernel_library(gumbel_softmax_grad_kernel DEPS ${COMMON_KERNEL_DEPS} softmax)
 kernel_library(math_kernel DEPS ${COMMON_KERNEL_DEPS} cast_kernel copy_kernel)
+kernel_library(reduce_kernel DEPS ${COMMON_KERNEL_DEPS} cast_kernel)
```
Comment (zyfncg) [inline paddle/phi/kernels/CMakeLists.txt:38]: cast_kernel按说可以自动解析依赖，这里是有特殊情况吗？

**Review by ZzSean — DISMISSED** [review]
LGTM for op benchmark

**Review by zyfncg — DISMISSED** [review]

File: paddle/phi/kernels/CMakeLists.txt (hunk 1 above)
Comment (MingMingShangTian) [reply paddle/phi/kernels/CMakeLists.txt:38]: 不加这句，编译时候找不到cast_kernel

**Review by YuanRisheng — DISMISSED** [review]

**Review by zyfncg — APPROVED** [review]

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

**Pull request merged**